### PR TITLE
Preparation work for development

### PR DESCRIPTION
**Pull request by alexmorley**
### Highlights

**Add continuous integration tests**
- Refactor tests so not all require GPU
- Set up CircleCI (might need to enable this in settings for a new repo)

**Pin environment and document set-up**
- Add versions (minor for some, patch for others) so we can break things rather than them being broken for us :) 

**Code review**
- TODO's added throughout the code. Higher-level summary and plan is in
https://github.com/alexmorley/pykilosort/issues/1 (FYI these are just thoughts about how things can be evolved, not criticisms of what's there currently!)

**Complete example script**
These scripts automatically download data from spikeforest so can be used in integration tests later on. They rely on spikeforest/spikeinterface (for now) but these aren't included in the main requirements so we can keep that minimal.

### CODE REVIEW

File: pykilosort/utils.py [hunk 1]
```diff
@@ -62,17 +62,19 @@ def _extend(x, i0, i1, val, axis=0):
 def is_fortran(x):
     if isinstance(x, np.ndarray):
         return x.flags.f_contiguous
-    raise ValueError()
+    raise ValueError('`is_fortran` is only implemented for numpy arrays')
 
 
 def _make_fortran(x):
+    # TODO: unclear - Are we using Dask? Are we explicitly support pykilosort for use with it?
```
Comment (rossant) [inline pykilosort/utils.py]: I started looking at dask but eventually abandoned it -- it's an awesome library but not the right tool for the job here. We can remove the reference here

File: pykilosort/utils.py [hunk 2]
```diff
@@ -62,17 +62,19 @@ def _extend(x, i0, i1, val, axis=0):
...
     if 'dask' in str(x.__class__):
         x = x.compute()
```
Comment (alexmorley) [inline pykilosort/utils.py]:
```suggestion
```

**Comment by rossant** [comment]
Hi @alexmorley , thanks for working on this! I don't have many comments. Is it ready to merge?

**Comment by alexmorley** [comment]
It is indeed. I like to leave things up for a day or two in case anyone has opinions :)

*Ideally* you could enable circleCI (free version) for this repository so that we can see the results of the tests every time someone makes a change. You should be able to sign up with your github account and add the project (only owners/admins can do this for repo's). The configuration is already included in this PR so no need to set anything else up.

**Pull request merged**